src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/Parser.cs

```diff
@@ -34,8 +34,19 @@ public Parser(ICommandHandler commandHandler, IVariableResolver variableResolver
             this.variableResolver = variableResolver;
             lexer = new Lexer(reader, name);
             tokenBuffer = new List<Token>();
+            DisableVariableSubstitution = variableResolver == null;
         }
 
+        /// <summary>
+        /// Whether to enable or disable the variable substitution.
+        /// A null IVariableResolver implies a true value.
```

Comment:
This isn't directly true though:

1. If `DisableVariableSubstitution` is false, and we don't have a variable resolver (or we have a variable resolver but it resolves some of the variables to null) and `ThrowOnUnresolvedVariable` is true then it will throw
2. But if `DisableVariableSubstitution` is true then it doesn't matter what `ThrowOnUnresolvedVariable` is - it'll parse just fine

Reply:
@Charles-Gagnon - what are you suggesting me to do here, besides adding a couple of tests to cover your 2 points there, which is usually a very effective way to answer questions :).

I was a bit thorn around the "should I do try to couple `ThrowOnUnresolvedVariable` and `DisableVariableSubstitution` even more and, essentially, go with  "setting DisableVariableSubstitution=true implies ThrowOnUnresolvedVariable=false". Is that what you are suggesting?

Reply:
I would say it makes sense to me to keep the existing behavior you have and just update the docs to clarify this. 

1. DisableVariableSubstitution I would just remove the bit about a null IVariableResolver - that doesn't really matter here and it should be clear enough that if a user is specifically disabling substitution it doesn't matter what the resolver is
2. ThrowOnUnresolvedVariable should mention that it only does this if DisableVariableSubstitution is false since that may not be quite as clear to me

And then of course test cases for those are appreciated 😄

Reply:
Cool. I'll have a couple of tests coming up + and updated comment.

Reply:
Cleaned up documentation. Added 8 little tests to cover the interaction of these properties.